### DISCUSSION TITLE
Any examples using a document as search input?

### DISCUSSION THREAD

**Discussion by ckmaresca** [marqo-ai/marqo#626] [Q&A]
I was looking through the docs, but I don't see a way to use a document instead of a string as search input.

I saw the note in a previous question about getting the vector of a document existing in db and using that for search but what about documents that don't exist in the db? 

I supposed you could upload a new document, get the `_id` and use that as the source, but I'm wondering if there is a simpler way?

Thx.

Chris.

**Comment by pandu-k** [comment]
Hi @ckmaresca, thanks for raising this issue! Yep we have a "search-by-doc" feature on our roadmap. But what do you mean by a document that doesn't exist? Users should be able to create queries that have the same affect as searching with an indexed document's content. 

If you have specific example it could help me understand your use case.

**Comment by ckmaresca** [comment]
Yes, search with a document that doesn't exist.   Right now as far as I can tell, it's only possible to search with a string rather than a multi-line document.

The reason for this is I tried pulling up an existing, stored document (as in your example) but it seems that insertion slices it up into a bunch of chunks and I am unsure how exactly to use this as a search query.  

It would be easier to just use a net-new document rather than inserting, reading stored document, re-assembling chunks (how?) and using that as the search query.

Thx.

Chris.
